### PR TITLE
[SCB-2089] 支持服务中心rbac认证

### PR DESCRIPTION
**Pull request by jeho0815**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/Cipher.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
```
Comment (liubao68) [inline foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/Cipher.java:3]: This class and implementation can move to foundation-common, do not need module foundation-auth

File: foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/TokenAuthHeaderProvider.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (liubao68) [inline foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/TokenAuthHeaderProvider.java:2]: This class and others move to huawei-cloud/servicestage module, do not need create a new module.

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java [hunk 3]
```diff
@@ -95,8 +95,8 @@
 
   private boolean isWatching = false;
 
-  private final ServiceLoader<AuthHeaderProvider> authHeaderProviders =
-      ServiceLoader.load(AuthHeaderProvider.class);
+  private final List<AuthHeaderProvider> authHeaderProviders =
+      SPIServiceUtils.getSortedService(AuthHeaderProvider.class);
```
Comment (liubao68) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java:99]: configure center is initialized before service center, will this cause problem?

File: foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/Cipher.java (hunk 1 above)
Comment (jeho0815) [reply foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/Cipher.java]: has removed to foundation-common

File: foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/TokenAuthHeaderProvider.java (hunk 2 above)
Comment (jeho0815) [reply foundations/foundation-auth/src/main/java/org/apache/servicecomb/foundation/auth/rbac/TokenAuthHeaderProvider.java]: has removed to registry-service-center module, they has more closer relationship

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java (hunk 3 above)
Comment (jeho0815) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterClient.java:99]: the dependency has some problem, we can refact later

**Pull request closed without merging**